cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/snowbridge_v2.rs

```diff
@@ -0,0 +1,1027 @@
+		type RuntimeEvent = <BridgeHubWestend as Chain>::RuntimeEvent;
+		let instructions = vec![
+			// Send message to Penpal
+			DepositReserveAsset {
+				// Send the token plus some eth for execution fees
+				assets: Definite(vec![eth_fee_penpal.clone(), token_asset].into()),
+				// Penpal
+				dest: Location::new(1, [Parachain(PARA_ID_B)]),
+				xcm: vec![
+					// Pay fees on Penpal.
+					PayFees { asset: eth_fee_penpal },
+					// Deposit assets to beneficiary.
+					DepositAsset {
+						assets: Wild(AllOf {
+							id: AssetId(token_location.clone()),
+							fun: WildFungibility::Fungible,
+						}),
+						beneficiary: beneficiary.clone(),
+					},
+					SetTopic(H256::random().into()),
+				]
+				.into(),
+			},
+		];
+		let xcm: Xcm<()> = instructions.into();
+		let versioned_message_xcm = VersionedXcm::V5(xcm);
+		let origin = EthereumGatewayAddress::get();
+
+		let message = Message {
+			origin,
+			assets,
+			xcm: versioned_message_xcm.encode(),
+			claimer: Some(claimer_bytes),
+			value: 3_500_000_000_000u128,
+			execution_fee: 1_500_000_000_000u128,
+			relayer_fee: 1_500_000_000_000u128,
+		};
+
+		let (xcm, _) = EthereumInboundQueueV2::do_convert(message, relayer_location).unwrap();
+		let _ = EthereumInboundQueueV2::send_xcm(xcm, AssetHubWestend::para_id().into()).unwrap();
+
+		assert_expected_events!(
+			BridgeHubWestend,
+			vec![RuntimeEvent::XcmpQueue(cumulus_pallet_xcmp_queue::Event::XcmpMessageSent { .. }) => {},]
+		);
+	});
+
+	AssetHubWestend::execute_with(|| {
+		type RuntimeEvent = <AssetHubWestend as Chain>::RuntimeEvent;
+		// Check that the assets were issued on AssetHub
+		assert_expected_events!(
+			AssetHubWestend,
+			vec![
+				RuntimeEvent::ForeignAssets(pallet_assets::Event::Issued { .. }) => {},
+				RuntimeEvent::XcmpQueue(cumulus_pallet_xcmp_queue::Event::XcmpMessageSent { .. }) => {},
+			]
+		);
+	});
+
+	PenpalB::execute_with(|| {
+		type RuntimeEvent = <PenpalB as Chain>::RuntimeEvent;
+
+		// Check that the token was received and issued as a foreign asset on PenpalB
+		assert_expected_events!(
+			PenpalB,
+			vec![RuntimeEvent::ForeignAssets(pallet_assets::Event::Issued { .. }) => {},]
+		);
+
+		// Beneficiary received the token transfer value
+		assert_eq!(
+			ForeignAssets::balance(token_location, AccountId::from(beneficiary_acc_bytes)),
+			token_transfer_value
+		);
+	});
+}
+
+#[test]
+fn send_foreign_erc20_token_back_to_polkadot() {
+	let relayer = BridgeHubWestendSender::get();
+	let relayer_location =
+		Location::new(0, AccountId32 { network: None, id: relayer.clone().into() });
+
+	let claimer = AccountId32 { network: None, id: H256::random().into() };
+	let claimer_bytes = claimer.encode();
+	let beneficiary =
+		Location::new(0, AccountId32 { network: None, id: AssetHubWestendReceiver::get().into() });
+
+	let asset_id: Location =
+		[PalletInstance(ASSETS_PALLET_ID), GeneralIndex(RESERVABLE_ASSET_ID.into())].into();
+
+	register_foreign_asset(eth_location());
```

Comment:
you should just make ETH (and maybe WETH) part of AH genesis for these tests and not have to register for every test: e.g. https://github.com/paritytech/polkadot-sdk/blob/698d9ae5b32785d3a5a55b770e973bbdb59ad271/cumulus/parachains/integration-tests/emulated/chains/parachains/assets/asset-hub-westend/src/genesis.rs#L78

Reply:
Added here: https://github.com/vgeddes/polkadot-sdk/blob/snowbridge-v2/cumulus/parachains/integration-tests/emulated/chains/parachains/assets/asset-hub-westend/src/genesis.rs#L102-L125